main/lsp/ExtractVariable.cc

```diff
@@ -55,7 +55,7 @@ class ExtractVariableWalk {
             // It's not valid to extract the following node types
             if (!ast::isa_tree<ast::Break>(tree) && !ast::isa_tree<ast::Next>(tree) &&
                 !ast::isa_tree<ast::Return>(tree) && !ast::isa_tree<ast::Retry>(tree) &&
-                !ast::isa_tree<ast::RescueCase>(tree)) {
+                !ast::isa_tree<ast::RescueCase>(tree) && !ast::isa_tree<ast::InsSeq>(tree)) {
```

Comment:
Instead of banning these, why not do something like

```
        } else if (insSeq->loc.contains(locOffsets)) {
            whereToInsert = insSeq->loc;
```

in the logic that computes `whereToInsert` if neither the `stats` nor the `expr` contained the loc?

Reply:
I can't write a test case for this because we don't have support for multi-line assertions, but the following is the issue that was cause crashes, where the selection is the entire method body.
```ruby
def foo
  1 + 1
  1 + 1
end
```
If we extracted this, we'd get something like:
```ruby
def foo
  newVariable = 1 + 1
  1 + 1
  newVariable
end
```
We could wrap the selection in parens and then collapse into a single line with `;`, but I think it's unlikely that people want to extract more than one line.

There's the separate question if people want to extract single line `InsSeq`s, but to me, it seems like enough of edge case that we just disable it until we get feedback to the opposite.

Reply:
Ah, I see, that makes sense. Can you note that in a comment in the test?